### PR TITLE
[consensus][easy] leverage commit_info to simplify code

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

Leverage commit_info to simplify some code.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

Existing tests.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: consensus/src/chained_bft/block_storage/block_tree.rs [hunk 1]
```diff
@@ -260,23 +257,10 @@ where
             .entry(block_id)
             .or_insert_with(|| Arc::clone(&qc));
 
-        let committed_block_id = qc.ledger_info().ledger_info().consensus_block_id();
```
Comment (davidiw) [inline consensus/src/chained_bft/block_storage/block_tree.rs]: this is some uncomfortable to look at code, very happy to see it cleaned up

File: consensus/src/chained_bft/block_storage/block_store_test.rs [hunk 2]
```diff
@@ -397,48 +397,6 @@ fn test_need_fetch_for_qc() {
     );
 }
 
-#[test]
```
Comment (davidiw) [inline consensus/src/chained_bft/block_storage/block_store_test.rs]: unclear why this test was removed

Comment (zekun000) [reply consensus/src/chained_bft/block_storage/block_store_test.rs]: the function now becomes very simple, and craft a LedgerInfo for the case needs additional helper function which I feel not worth the effort, but let me know your thought

File: consensus/src/chained_bft/block_storage/block_store_test.rs [hunk 3]
```diff
@@ -397,48 +397,6 @@ fn test_need_fetch_for_qc() {
...
-    let qc = placeholder_certificate_for_block(
-        vec![inserter.signer()],
-        HashValue::zero(),
-        a3.round() + 3,
-        HashValue::zero(),
-        a3.round() + 2,
-        None,
-    );
-    assert_eq!(
-        block_store.need_sync_for_quorum_cert(HashValue::zero(), &qc),
```
Comment (davidiw) [inline consensus/src/chained_bft/block_storage/block_store_test.rs]: so if you used this function without the hashvalue::zero it wouldn't work as expected?

Comment (zekun000) [reply consensus/src/chained_bft/block_storage/block_store_test.rs]: it hand crafted the certified round and the committed block id, but now we can directly use the committed round so the test case doesn't make much sense

**Pull request closed without merging**